src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -26,6 +26,16 @@ export function ThemeProvider({
     [unstableGlobalTheming, themeConfig],
   );
 
+  // We want these values to be `null` instead of `undefined` when not set.
+  // Otherwise, setting a style property to `undefined` does not remove it from the DOM.
+  const backgroundColor = customProperties['--p-surface-background'] || null;
```

Comment:
This will be better to grab from context once we make the colors available. Something like `theme.surfaceBackground` so if the color role name changes the type check will fail. Otherwise, if we change the color role name now, nothing would fail.